api/handler/put.go

```diff
@@ -206,15 +206,17 @@ func (h *handler) PutObjectHandler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	settings, err := h.obj.GetBucketSettings(r.Context(), bktInfo)
-	if err != nil {
+	if err != nil && !errors.IsS3Error(err, errors.ErrNoSuchKey) {
 		h.logAndSendError(w, "could not get bucket settings", reqInfo, err)
 		return
 	}
 
-	params.Lock, err = formObjectLock(bktInfo, settings.LockConfiguration, r.Header)
-	if err != nil {
-		h.logAndSendError(w, "could not form object lock", reqInfo, err)
-		return
+	if settings != nil {
+		params.Lock, err = formObjectLock(bktInfo, settings.LockConfiguration, r.Header)
```

Comment:
Actually, we have to check `bktInfo.ObjectLockEnabled` (it's done in `formObjectLock` function)
```suggestion
	if settings == nil {
	        settings = &data.BucketSettings{}
	}
```